### PR TITLE
test: move protocol specs to main process

### PR DESCRIPTION
**Pull request by nornagon**
#### Description of Change
This moves the tests for `protocol`, which is a main process feature, to the main process.

I also expanded a couple of the `registerSchemesAsPrivileged` tests to test XHR as well as fetch, and check console messages.

#### Checklist

- [x] PR description included and stakeholders cc'd
- [x] `npm test` passes
- [x] tests are [changed or added](https://github.com/electron/electron/blob/master/docs/development/testing.md)
- [x] relevant documentation is changed or added
- [x] PR title follows semantic [commit guidelines](https://github.com/electron/electron/blob/master/docs/development/pull-requests.md#commit-message-guidelines)
- [x] [PR release notes](https://github.com/electron/clerk/blob/master/README.md) describe the change in a way relevant to app developers, and are [capitalized, punctuated, and past tense](https://github.com/electron/clerk/blob/master/README.md#examples).

#### Release Notes

Notes: none

### CODE REVIEW

File: docs/api/protocol.md [hunk 1]
```diff
@@ -342,11 +346,12 @@ which sends a file as a response.
 * `handler` Function
   * `request` Object
     * `url` String
+    * `headers` Object
     * `referrer` String
     * `method` String
     * `uploadData` [UploadData[]](structures/upload-data.md)
   * `callback` Function
-    * `data` String (optional)
+    * `data` (String | Object) (optional)
```
[the remark this answers is not in the document]
Comment (nornagon) [reply docs/api/protocol.md]: I expanded the real structure instead

File: spec-main/api-protocol-spec.ts [hunk 2]
```diff
@@ -1,125 +1,111 @@
...
+    setTimeout(resolve, ms)
+  })
+}
+
+function getStream (chunkSize = text.length, data: Buffer | string = text) {
+  const body = new stream.PassThrough()
+
+  async function sendChunks () {
+    await delay(0)  // the stream protocol API breaks if you send data immediately.
+    let buf = Buffer.from(data as any)  // nodejs typings are wrong, Buffer.from can take a Buffer
```
[the remark this answers is not in the document]
Comment (nornagon) [reply spec-main/api-protocol-spec.ts]: /shrug, nodejs typings should reflect that then. `Buffer.from(Buffer.from("foo"))` works in nodejs but the types reject it.

File: spec-main/api-protocol-spec.ts [hunk 3]
```diff
@@ -1,125 +1,111 @@
...
+  return new Promise((resolve) => {
+    setTimeout(resolve, ms)
+  })
+}
+
+function getStream (chunkSize = text.length, data: Buffer | string = text) {
+  const body = new stream.PassThrough()
+
+  async function sendChunks () {
+    await delay(0)  // the stream protocol API breaks if you send data immediately.
```
[the remark this answers is not in the document]
Comment (nornagon) [reply spec-main/api-protocol-spec.ts]: I'm guessing the nws13n of the protocol api will fix it.

File: spec-main/api-protocol-spec.ts [hunk 4]
```diff
@@ -565,21 +513,21 @@ describe('protocol module', () => {
...
         const data = {
           url: url,
           method: 'POST',
           uploadData: {
             contentType: 'application/x-www-form-urlencoded',
-            data: request.uploadData[0].bytes.toString()
+            bytes: request.uploadData[0].bytes
           },
-          session: null
+          session: null as any
```
Comment (MarshallOfSound) [inline spec-main/api-protocol-spec.ts]: Why does this need to be casted to any? If it's an allowed value lets update the docs

Comment (nornagon) [reply spec-main/api-protocol-spec.ts]: looks like it doesn't /shrug

this test is disabled, btw. not sure why.

**Pull request merged**